### PR TITLE
:bug:  Source Zendesk Support: fix incremental logic for ticket_comments stream

### LINKED ISSUE #5695: Zendesk Support Connector successful sync but result in retry
This is a follow-up ticket for #5569 for specifically Zendesk Support connector.

## Enviroment
- **Airbyte version**:  0.29.11-alpha
- **OS Version / Instance**: Linux EC2 m5.2xlarge
- **Deployment**: Docker
- **Source Connector and version**: Zendesk Support Native 0.1.0
- **Destination Connector and version**: Redshift 0.3.12
- **Severity**: Critical
- **Step where error happened**: Sync

## Current Behavior
Successful job ends with retry

## Expected Behavior
Successful job should end with success and no retry

## Logs
From the user: `first failed sync (actually passes) and the retry actually failed`

image
[zendesk_succeed.txt](https://github.com/airbytehq/airbyte/files/7080327/zendesk_succeed.txt)
[zendesk_fail.txt](https://github.com/airbytehq/airbyte/files/7080328/zendesk_fail.txt)

## Steps to Reproduce
1. Create Zendesk Support connector
2. Run
3. See retries

## Are you willing to submit a PR?
No.

### PR DESCRIPTION
**Pull request by antixar**
## What
For loading of all comments this connector needs a list of all updated tickets. But the stream tickets have different cursor formats e.g:  
```
Ticket state "generated_timestamp": 1627637409}
Comment state: {"created_at": "2021-07-30T12:30:09Z"}
```
the old conversion between them was incorrect because the connector had to know a user timezone.

## Recommended reading order
1. `streams.py`

## Pre-merge Checklist

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (midavadim) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:82]: can it be lower that 0.  f not then just simple check is enough 'If retry_after then'

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (midavadim) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:88]: can it be lower that 0.  f not then just simple check is enough 'If retry_after then'

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (midavadim) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:164]: why do we need to return None instead of current stream state ?

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:87]: Check comment above

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:82]: It can be as fact. There is an[ issue](https://github.com/airbytehq/airbyte/issues/5569) about this

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:88]: add a same comment before

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:164]: not need. it was testing some assumption only. returned the standard behaviour

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:81]: we need to check the presence of several headers: Retry-After or X-Rate-Limit. Thus for the first checking we can't set the default value.

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:87]: agree for this code part. Corrected

**Review by lazebnyi — APPROVED** [review]

**Review by midavadim — COMMENTED** [review]
What is the main part of the update which solves the problem?

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:165]: agree. corrected

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/users.json
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/users.json]: These fields have optional values. And updated CI was checking them incorrectly. This CI issue was fixed some days ago and I've restored previous format values.

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:89]: I know this isn't a change from this PR but looks like you edited some stuff here so worth asking...

What's the reason for default rate limit being 1 request per minute? That seems especially slow

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json
Comment (Phlair) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json:134]: not date-time anymore?

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (Phlair) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:225]: this comment looks like it relates to the old code?

**Review by Phlair — COMMENTED** [review]
lgtm, I'll wait for tests to pass before approve.

Just some very minor comments

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:89]: I'm not sure that necessary headers are always exists. Ok, I've corrected this default value and it would return the super value.

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json:134]: there was a CI bug with optional datetime fields. But it was fixed. I've restored as was

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:225]: Fixed

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json
Comment (Phlair) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json:135]: does this need to be `date-time`?

**Review by Phlair — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json
Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json:135]: /publish connector=source-zendesk-support

Comment (antixar) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/schemas/ticket_metrics.json:135]: fixed

**Pull request merged**
